app/views/devise/shared/_password_strength.html.erb

```diff
@@ -10,9 +10,9 @@
       <span class='h6'>
         <%= t('instructions.password.strength.intro') %>
       </span>
-      <span id='pw-strength-txt' class='bold' data-forbidden-passwords='<%= @forbidden_passwords %>'>
-        ...
-      </span>
+      <%= tag.span '...', id: 'pw-strength-txt', class: 'bold', data: {
+        'forbidden-passwords': @forbidden_passwords,
```

Comment:
> Open Questions: Is there a better way to ensure consistency of or consolidate how the password edit view is prepared? Similarly, there appear to be a few different variations of users (token_user, build_user, resource) within the ResetPasswordsController class, not all of which are available to reference in every method handler.

One thing we could do is have the template raise if `@forbidden_passwords` is nil, defined?
I don't like this but it would definitely mean we find out sooner rather than later....

```suggestion
        'forbidden-passwords': @forbidden_passwords.tap { |p| raise "missing forbidden_passwords" unless p },
```

Reply:
> One thing we could do is have the template raise if `@forbidden_passwords` is nil, defined?
> I don't like this but it would definitely mean we find out sooner rather than later....

Hmm. I think this would help give us visibility where disparate setup logics fail to assign the expected view variable, but I'd worry if the instinct to resolve that would be to patch where it's missing, vs. to avoid having multiple setups in the first place.

I was thinking more along the lines of: Could we redirect the user to the edit handler? Or otherwise reuse the logic of the `edit` method where this is set up initially?

Reply:
Another option would be to add a helper method like `load_forbidden_passwords` which would:

1. raise an error if undefined
2. highlight a need to include the helper in the controller class so it's available?

Reply:
> Another option would be to add a helper method like `load_forbidden_passwords` which would:

Part of the challenge with this is that the source of those values often changes (e.g. user lookup by token, etc).

One minor improvement I considered is to avoid referencing the controller variable from the partial, which I've read is a common Rails anti-pattern obscuring where these variables are referenced.

In 2d0c49af1, I revised to pass the value to the partial from the controller view, which gives a bit more transparency to where we're expecting it to be assigned. Notably, in doing this, I noticed another case where we weren't passing the value: In the event disavowal form. This is resolved in 5bae171fd.

I was curious why the event disavowal path wasn't one of the original routes included in the error being resolved here (LG-3738). It turns out that there's another error preventing it from getting to the point where the original reported error would occur 😅   (["Cannot read property 'addEventListener' of null"](https://github.com/18F/identity-idp/blob/977a1d8070592864450a906275d24137002eca8e/app/javascript/packs/pw-strength.js#L134)). This is fixed in 2271b9bf5.

It's improved now, though this all seems to reinforce the fact that manually preparing the variable (and hard-coded references to password input IDs) is rather brittle.

Reply:
Using a local variable to the partial seems like a good compromise